internal/scheduler/preempting_queue_scheduler.go

```diff
@@ -861,6 +861,9 @@ func defaultPostEvictFunc(ctx context.Context, job interfaces.LegacySchedulerJob
 	}
 
 	// Add a toleration to allow the job to be re-scheduled even if node is unschedulable.
+	//
+	// TODO: Because req is created with a new tolerations slice above, this toleration doesn't persist.
```

Comment:
This is a bit unfortunate. But I don't see a good solution until we have a common job notion that doesn't create new objects every time.